forwards_paper.tex

```diff
@@ -656,6 +661,26 @@ \subsection*{Recording the pedigree in forwards time}
 ancestors in the current population, and so the entire path tracing back to
 its common ancestor with 27 is redundant.
 
+\paragraph{Essential steps in tree sequence recording}
+To record a tree sequence, including mutations,
+a simulator must record:
+\begin{itemize}
+    \item the birth time of each individual in the Node Table,
+    \item the genetic segments each individual inherits from her parents in the Edge Table, and
+    \item the individual each new mutation occurs in, and the derived state in the Mutation Table,
+        adding also to the Site Table if the site is previously unmutated.
+\end{itemize}
+Recall that here we mean ``\emph{haploid} individual'', i.e., gamete.
+Before simplification, time must be translated to ``time ago'' if it is not already,
+and the tables must be sorted.
+Algorithm~\algref{W} avoided both of these details since $t$ denotes
+``time until the end of the simulation'',
+and nonoverlapping generations ensure that the tables will be already sorted.
```

Comment:
This isn't true, unfortunately. If we want the edges to be sorted, the we need to buffer the edges for each generation, and output then in order of parent ID. We just ignored this complication in Alg W.

Reply:
Ah, right: [sorted on parent id](https://github.com/jeromekelleher/msprime/blob/master/msprime/tables.py#L1191).

BTW, [the documentation](http://msprime.readthedocs.io/en/latest/interchange.html) currently doesn't quite say this:

> All edges for a given parent must be contiguous;
> Edges must be listed in nondecreasing order of parent time;
> Within the edges for a given parent, edges must be sorted first by child ID and then by left coordinate.

Reply:
> Ah, right: sorted on parent id.

No, not quite. We don't require them to be sorted by parent ID, we just require all edges for a given parent to be contiguous *and* (overall) be listed in nondecreasing order of parent time. So my statement above wasn't quite accurate: we need to buffer the edges for a given parent **until that parent** dies, and then flush them. This will satisfy the requirements I think.

Reply:
Yep; I'm on board now.